content/courses/advanced-ada/chapters/aggregates.rst

```diff
@@ -1196,6 +1196,865 @@ Later on, we'll discuss
+    begin
+       --  Positional component association
+       M := [[0, 1, 2],
+             [3, 4, 5]];
+
+       Display (M);
+
+       --  Named component association
+       M := [[1 => 3,
+              2 => 4,
+              3 => 5],
+             [1 => 6,
+              2 => 7,
+              3 => 8]];
+
+       Display (M);
+
+    end Show_Array_Aggregates;
+
+The first aggregate we use in this example is :ada:`[[0, 1, 2], [3, 4, 5]]`.
+Here, :ada:`[0, 1, 2]` and :ada:`[3, 4, 5]` are subaggregates of the
+multidimensional aggregate. Subaggregates don't have a type themselves, but are
+rather just considered part of a multidimensional aggregate (which, of course,
+has an array type). In this sense, a subaggregate such as :ada:`[0, 1, 2]` is
+different from a one-dimensional aggregate (such as :ada:`[0, 1, 2]`), even
+though they are written in the same way.
+
+Strings in subaggregates
+^^^^^^^^^^^^^^^^^^^^^^^^
+
+In the case of matrices using characters, we can use strings in the
+corresponding array aggregates. Consider this package:
+
+.. code:: ada compile_button project=Courses.Advanced_Ada.Aggregates.String_Aggregates
+
+    package String_Lists is
+
+       type String_List is array (Positive range <>,
+                                  Positive range <>)
+                                  of Character;
+
+       procedure Display (SL : String_List);
+
+    end String_Lists;
+
+    pragma Ada_2022;
+
+    with Ada.Text_IO; use Ada.Text_IO;
+
+    package body String_Lists is
+
+       procedure Display (SL : String_List) is
+
+          procedure Display_Row (SL : String_List;
+                                 I  : Integer) is
+          begin
+             Put ("  (");
+             for J in SL'Range (2) loop
+                Put (SL (I, J));
+             end loop;
+             Put (")");
+          end Display_Row;
+
+       begin
+          Put_Line ("Length (1) = "
+                    & SL'Length (1)'Image);
+          Put_Line ("Length (2) = "
+                    & SL'Length (2)'Image);
+
+          Put_Line ("(");
+          for I in SL'Range (1) loop
+             Display_Row (SL, I);
+             if I /= SL'Last (1) then
+                Put_Line (",");
+             else
+                New_Line;
+             end if;
+          end loop;
+          Put_Line (")");
+       end Display;
+
+    end String_Lists;
+
+Then, when assigning to an object :ada:`SL` of :ada:`String_List` type, we can
+use strings in the aggregates:
+
+.. code:: ada run_button project=Courses.Advanced_Ada.Aggregates.String_Aggregates
+
+    pragma Ada_2022;
+
+    with String_Lists; use String_Lists;
+
+    procedure Show_Array_Aggregates is
+       SL : String_List (1 .. 2, 1 .. 3);
+    begin
+       --  Positional component association
+       SL := ["ABC",
+              "DEF"];
+
+       Display (SL);
+
+       --  Named component association
+       SL := [[1 => 'X',
+               2 => 'Y',
+               3 => 'Z'],
```

Comment:
Consider using the same character components here is the name association version, in other words ABC and DEF, so that the reader can more easily see the differences from the positional form.

Reply:
I originally wanted to have an example with `others` as well. However, I agree with your reasoning, so I've just added the equivalent code (that initializes `"ABC","DEF"`) as you suggested.